src/v/security/audit/schemas/utils.cc

```diff
@@ -0,0 +1,147 @@
+#include "security/audit/schemas/types.h"
+#include "utils/request_auth.h"
+
+#include <boost/algorithm/string/predicate.hpp>
+
+namespace security::audit {
+
+namespace {
+
+api_activity::activity_id http_method_to_activity_id(std::string_view method) {
+    return string_switch<api_activity::activity_id>(method)
+      .match_all("POST", "post", api_activity::activity_id::create)
+      .match_all("GET", "get", api_activity::activity_id::read)
+      .match_all("PUT", "put", api_activity::activity_id::update)
+      .match_all("DELETE", "delete", api_activity::activity_id::delete_id)
+      .default_match(api_activity::activity_id::unknown);
+}
+
+uniform_resource_locator
+uri_from_ss_http_request(const ss::http::request& req) {
+    return {
+      .hostname = req.get_header("Host"),
+      .path = req.format_url(),
+      .port = port_t{req.get_client_address().port()},
+      .scheme = req.get_protocol_name(),
+      .url_string = req.get_url()};
+}
+
+http_request from_ss_http_request(const ss::http::request& req) {
+    using ss_http_headers_t = decltype(req._headers);
+    using ss_http_headers_value_t = ss_http_headers_t::value_type;
+    const auto get_headers = [](const ss_http_headers_t& headers) {
+        std::vector<http_header> audit_headers;
+        std::transform(
+          headers.begin(),
+          headers.end(),
+          std::back_inserter(audit_headers),
+          [](const auto& kv) {
+              const auto sanitize_header =
+                [](const ss_http_headers_value_t& kv) -> ss::sstring {
+                  static const std::vector<std::string_view> sensitive_headers{
+                    "Authorization"};
+
+                  if (
+                    std::find_if(
+                      sensitive_headers.begin(),
+                      sensitive_headers.end(),
+                      [&kv](std::string_view s) {
+                          return boost::iequals(s, kv.first);
+                      })
+                    != sensitive_headers.end()) {
+                      return "******";
+                  } else {
+                      return kv.second;
+                  }
+              };
+              return http_header{
+                .name = kv.first, .value = sanitize_header(kv)};
+          });
+        return audit_headers;
+    };
+    return http_request{
+      .http_headers = get_headers(req._headers),
+      .http_method = req._method,
+      .url = uri_from_ss_http_request(req),
+      .user_agent = req.get_header("User-Agent"),
+      .version = req._version};
+}
+
+network_endpoint from_ss_endpoint(const ss::socket_address& sa) {
+    return network_endpoint{
+      .addr = net::unresolved_address(
+        fmt::format("{}", sa.addr()), sa.port(), sa.addr().in_family())};
+}
+
+/// TODO: Via ACLs metadata return correct response
+api_activity_unmapped unmapped_data() {
+    return api_activity_unmapped{.shard_id = ss::this_shard_id()};
```

Comment:
Did we decide to remove this because of YAGNI?

Reply:
I'm not opposed to removing it.  There's a timestamp field that can be used for maintaining order.